File: backend/src/main/java/com/talenthub/service/AuditionService.java
package com.talenthub.service;

import com.talenthub.dto.AuditionRequestDto;
import com.talenthub.dto.AuditionResponseDto;
import java.util.List;

public interface AuditionService {
    AuditionResponseDto createAudition(Long organizerUserId, AuditionRequestDto dto);
    List<AuditionResponseDto> getMyAuditions(Long organizerUserId);
    AuditionResponseDto updateAudition(Long organizerUserId, Long auditionId, AuditionRequestDto dto);
    void deleteAudition(Long organizerUserId, Long auditionId);
    List<AuditionResponseDto> getActiveAuditions();
    AuditionResponseDto getAuditionDetails(Long auditionId);
}

File: frontend/src/app/core/services/audition.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class AuditionService {
  private apiUrl = `${environment.apiUrl}/auditions`;

  constructor(private http: HttpClient) {}

  getActiveAuditions(): Observable<any[]> {
    return this.http.get<any[]>(this.apiUrl);
  }

  getAuditionDetails(id: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/${id}`);
  }

  getMyAuditions(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/my`);
  }

  createAudition(audition: any): Observable<any> {
    return this.http.post<any>(this.apiUrl, audition);
  }

  updateAudition(id: number, audition: any): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/${id}`, audition);
  }

  deleteAudition(id: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${id}`);
  }
}
